source/lib.js

```diff
@@ -546,10 +546,20 @@ function processCallMemberExpression(node) {
   for (let i = 0; i < children.length; i++) {
     const glob = children[i]
     if (glob?.type === "PropertyGlob") {
-      // TODO: add ref to ensure object base evaluated only once
-      const prefix = children.slice(0, i)
-        .concat(glob.dot)
+      let prefix = children.slice(0, i)
       const parts = []
+      let hoistDec
+      // add ref to ensure object base evaluated only once
+      if (prefix.length > 1) {
+        const ref = {
+          type: "Ref",
+          base: "ref",
+        }
+        hoistDec = [["", ["const ", ref, " = ", prefix], ";"]]
+        prefix = [ref]
+      }
+      prefix = prefix.concat(glob.dot)
```

Comment:
I think you could just do `prefix.push(glob.dot)` here because in all cases `prefix` is a fresh array.

Reply:
`prefix` is a shared ref in the `hoistDec` array so we need a clean ref to not mutate it.